pkg/virt-controller/services/template.go

```diff
@@ -168,6 +169,67 @@ func CPUModelLabelFromCPUModel(vmi *v1.VirtualMachineInstance) (label string, er
 	return
 }
 
+func CPUFeatureLabelsFromCPUFeatures(vmi *v1.VirtualMachineInstance) []string {
+	var labels []string
+	if vmi.Spec.Domain.CPU != nil && vmi.Spec.Domain.CPU.Features != nil {
+		for _, feature := range vmi.Spec.Domain.CPU.Features {
+			if feature.Policy == "" || feature.Policy == "require" {
+				labels = append(labels, NFD_CPU_FEATURE_PREFIX+feature.Name)
```

Comment:
Should we sanitize the cpufamily / feature names (i.e. make them lowercase)?

IIRC then Kubernetes is also lowercasing all labels.

Reply:
> IIRC then Kubernetes is also lowercasing all labels.
lowercasing at what stage ?  , in this PR the features/models are translated to nodeselectors (not just labels in their raw form) .
The nodeselector's labels are not lower-cased when added to the virt-launcher pod (creating a vmi with a spec that has Uppercase labels under a node selector will translate to uppercase labels in virt-launcher)

Reply:
the valid [label syntax](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) includes upper-case letters.

Reply:
Right. I just think that I noticed that nfd exposed cpu labels in lower-case

Reply:
But that's not blocking, was just a perception.